exercises/intro/intro2.d

```diff
@@ -1,3 +1,6 @@
+/// Uh-oh, the program does not compile!
+/// Solve it by making the code print a greeting!
+
 import std.stdio;
 
 // Just D it
```

Comment:
This is just a minor point, but I feel it's too much on this exercise to introduce Dlang's UFCS already. A beginner might be confused with this syntax, and not understand where exactly an argument should be placed.

I'd change this exercise to require users to solve it in a familiar syntax (`writefln("Hello %s", );`) first, and then create another exercise (intro3.d) with the exact same challenge, but with this current form (`writefln!"Hello %s"();`), designed to teach users the UFCS concept (or postpone it further if you feel it should be given more attention)

Reply:
This is not UFCS though, it's a template call. I chose to use the template version to pass the format into because it asserts at compile time, leading to a clearer error message, instead of exploding with an exception at run time.

Reply:
Right, but this version (`writefln("Hello %s", );`) should still result in a compile-time error (?).

Reply:
No, the format is a separate thing. Only the template version verifies the format at compile time. That version bypasses this verification, resulting only in a run time exception.